pkgs/applications/misc/blender/default.nix

```diff
@@ -71,7 +71,7 @@
   pkg-config,
   potrace,
   pugixml,
-  python3Packages, # must use instead of python3.pkgs, see https://github.com/NixOS/nixpkgs/issues/211340
+  python311Packages, # must use instead of python3.pkgs, see https://github.com/NixOS/nixpkgs/issues/211340
```

Comment:
So Blender is not compatible with Python 3.12?

Reply:
Blender runs fine on `python3.12` This was changed in #325292, too. If we don't follow `python3Packages` we introduce errors between the python versions (The ones actually fixed by that PR)

Reply:
> Blender runs fine on `python3.12`

As pointed out, that doesn't mean addons do.

Reply:
See the comment by @kirillrdy (https://github.com/NixOS/nixpkgs/pull/325292#discussion_r1667893187)

Reply:
> If we don't follow `python3Packages` we introduce errors between the python versions

Not once this PR or https://github.com/NixOS/nixpkgs/pull/326041 is merged.

Reply:
> See the comment by @kirillrdy ([#325292 (comment)](https://github.com/NixOS/nixpkgs/pull/325292#discussion_r1667893187))

That's just personal preference regarding style and has nothing to do with whether Python 3.11 or 3.12 should be used.

Reply:
I had to update the previous `python311Packages` to `python312Packages` in order to not have a version mismatch. The changes to `python3Packages` were not only cosmetic, but also functional, because we don't have to change the version by hand after each python upgrade. 
I agree, once #326041 is merged, this shouldn't matter :laughing:

Reply:
@dotlambda If I understand correctly, blender probably should be using python 3.11 due to some plugins still not compatible with 3.12. As for recommendation to change default versions in all-packages.nix or python-packages.nix, I see it as official recommendation rather than person preference, please correct me otherwise.

Reply:
> I see it as official recommendation rather than person preference

Reference?

Reply:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md#changing-implicit-attribute-defaults